clients/src/test/java/org/apache/kafka/clients/admin/MockAdminClient.java

```diff
@@ -330,6 +331,12 @@ synchronized public DescribeTopicsResult describeTopics(Collection<String> topic
                 future.completeExceptionally(new UnknownTopicOrPartitionException("Topic " + requestedTopic + " not found."));
                 topicDescriptions.put(requestedTopic, future);
             }
+            // try to simulate the leader not available situation when topic name is "LeaderNotAvailableTopic"
+            if (requestedTopic.equals("LeaderNotAvailableTopic")) {
+                KafkaFutureImpl<TopicDescription> future = new KafkaFutureImpl<>();
+                future.completeExceptionally(new LeaderNotAvailableException("The leader of Topic " + requestedTopic + " is not available."));
```

Comment:
Try to simulate the `LeaderNotAvailableException` in the `MockAdminClient`, by setting the topic name to `LeaderNotAvailableTopic`.

Reply:
Is it possible to use `EasyMock` instead of adding this to the actual `MockAdminClient`? I know it's kind of a pain to set up but I think it'll make the test a lot more clear. I did something similar in StreamsPartitionAssignorTest to mock the results of the `listOffsets` request